go/vt/vtorc/db/db.go

```diff
@@ -160,36 +126,27 @@ func deployStatements(db *sql.DB, queries []string) error {
 func initVTOrcDB(db *sql.DB) error {
 	log.Info("Initializing vtorc")
 	log.Info("Migrating database schema")
-	_ = deployStatements(db, generateSQLBase)
-	_ = deployStatements(db, generateSQLPatches)
+	_ = deployStatements(db, vtorcBackend)
 	_ = registerVTOrcDeployment(db)
 
-	if IsSQLite() {
-		_, _ = ExecVTOrc(`PRAGMA journal_mode = WAL`)
-		_, _ = ExecVTOrc(`PRAGMA synchronous = NORMAL`)
-	}
+	_, _ = ExecVTOrc(`PRAGMA journal_mode = WAL`)
```

Comment:
I am not sure I follow what you are suggesting. db is an external struct `db *sql.DB`. I could create a method which initializes a db, but that would do a similar check wouldn't it? Yes, you are right we would have to re-introduce these checks if we decide to support more backing stores. 

The correct way to do it in my opinion would be to do something like this - 
```go

type VTOrcBackend struct {
	db     *sql.DB
	dbImpl dbImplementation
}

type dbImplementation interface {
	Init(db *sql.DB)
}

type SQLite3Impl struct {
}

func (s *SQLite3Impl) Init(db *sql.DB) {
	_, _ = ExecVTOrc(`PRAGMA journal_mode = WAL`)
	_, _ = ExecVTOrc(`PRAGMA synchronous = NORMAL`)
}

func initVTOrcBackend(db *sql.DB) *VTOrcBackend {
	if IsSQLite() {
		backend := &VTOrcBackend{
			db:     db,
			dbImpl: &SQLite3Impl{},
		}
		backend.dbImpl.Init(backend.db)
	} else {
		....	
	}
}
```

We can them add execute and other functions if there implementations are different. This architecture didn't exist in VTOrc before either. There were just a bunch of extensive `IsSQLLite` checks everywhere. In this PR I have cleaned it all up since we don't support any other backing store currently. If we do decide to add one, I would like to do it this way, but I don't want to spend time making this change now since we don't have anything to gain from it.

Reply:
Agree keeping it like this right now and then we can change the design in future if needed.